src/backend/metal/src/device.rs

```diff
@@ -1351,8 +1349,8 @@ impl hal::Device<Backend> for Device {
                     vertex_buffers.alloc().init((original.clone(), base_offset));
                     vertex_buffers.len() - 1
                 });
-            let mtl_buffer_index = attribute_buffer_index as usize + relative_index;
-            if mtl_buffer_index >= self.shared.private_caps.max_buffers_per_stage as usize {
+            let mtl_buffer_index = self.shared.private_caps.max_buffers_per_stage - 1 - (relative_index as ResourceIndex);
+            if mtl_buffer_index < pipeline_layout.total.vs.buffers {
```

Comment:
Could you explain this condition? I expected to test `mtl_buffer_index < 0` here but maybe I'm misunderstanding

Reply:
We are allocating the vertex buffers from top to bottom. It starts at 31, then goes down, but we can't get lower than `pipeline_layout.total.vs.buffers` since this is where the pipeline layout bindings (for resources and push constants) end.

Reply:
:+1: This makes sense, thanks! I thought we were also including the vertex buffers in that count but I see the distinction now (for the pipeline layout)